scheduler-worker/src/test/java/gov/nasa/jpl/aerie/scheduler/worker/services/SchedulingIntegrationTests.java

```diff
@@ -1682,6 +1686,59 @@ export default () => Goal.CoexistenceGoal({
     assertEquals(Duration.of(65, Duration.MINUTES), peelBanana.startOffset());
   }
 
+  @Test
+  void testOptionalSimulationAfterGoal() {
```

Comment:
How does this prove that there was no simulation between the goals? Wouldn't a more accurate test be one where, by not simulating between the goals, we get a different result set than what we'd get if we _had_ simulated between each goal?

Reply:
That's exactly what it does. I can add a comment explaining how though.

Reply:
(There was a comment already there, but that was a mistake, I must have copy-pasted from somewhere else)

Reply:
This test makes sure that resources are stale. I've added a second test that makes sure it doesn't care about directives that don't have corresponding spans. I've also modified both of them to make sure the results are different, compared to if it *is* resimulated.